services/logger/listener/src/lib/Logger.js

```diff
@@ -0,0 +1,45 @@
+function Logger({ sns, targetArn, emitter }) {
```

Comment:
Nothing agains a function and a prototype syntax, but i'm confused - why not a class?
Nice architectural approach anyway!

Reply:
That old debate about prototypes vs classes 😆 In this case, it's just a personal preference

https://github.com/joshburgess/not-awesome-es6-classes

Reply:
I understand, I just couldn't help but comment it for you especially when you have all the latest features of the languages. It's a simple scenario without inheritance or anything crazy, a bit of syntax sugar wouldn't hurt in the same way as async/await.

Reply:
Here's how the sugar would look like:
```
class Logger {
  constructor({ sns, targetArn, emitter }) {
    this.sns = sns;
    this.targetArn = targetArn;
    this.emitter = emitter;
  }

  log(
    {
      level = 'info',
      type = 'file',
      message = '',
    }) {
    return this.sns
      .publish({
        Message: JSON.stringify({
          default: JSON.stringify({
            emitter: this.emitter,
            level,
            type,
            time: new Date().toISOString(),
            message,
          }),
        }),
        MessageStructure: 'json',
        TargetArn: this.targetArn,
      })
      .promise();
  }

  info({ type = 'file', message = '' }) {
    return this.log({
      level: 'info',
      type,
      message,
    });
  }

  error({ type = 'file', message = '' }) {
    return this.log({
      level: 'error',
      type,
      message,
    });
  }
}

export default Logger;

```